src/api/PCOL/types.ts

```diff
@@ -1,19 +1,27 @@
 import { z } from "zod";
 
-const NewPCOL = z.object({
-  Subject: z.string(),
-  References: z.string(),
-  DODAAC: z.string(), //lookup
-  Contract: z.string(), //lookup
-  DeliveryOrderMod: z.string(),
-  RFI: z.string(),
-  RFP: z.string(),
-  ECP: z.string(),
-  CCP: z.string(),
-  AssociatedContractorLetterNumbers: z.string(),
-  Disclaimers: z.array(z.string()),
-  CarbonCopy: z.string(),
-  AdditionalDistributionInfo: z.string(),
-});
+const NewPCOL = z
+  .object({
+    Subject: z.string(),
+    References: z.string(),
+    DODAAC: z.string(), //lookup
+    Contract: z.string(), //lookup
+    Contractor: z.string().optional(), //lookup
+    ContractorPOC: z.string(), //lookup
+    DeliveryOrderMod: z.string(),
+    RFI: z.string(),
+    RFP: z.string(),
+    ECP: z.string(),
+    CCP: z.string(),
+    AssociatedContractorLetterNumbers: z.string(),
+    Disclaimers: z.array(z.string()),
+    CarbonCopy: z.string(),
+    AdditionalDistributionInfo: z.string(),
+  })
+  .refine(
+    (pcol) =>
+      pcol.Contract !== "No Established Contract" ||
+      (pcol.Contractor !== undefined && pcol.Contractor.length >= 3)
+  );
 
 export type NewPCOL = z.infer<typeof NewPCOL>;
```

Comment:
Not related to this line, but wanted to log it on the PR so we don't forget. SharePoint Column Validation is not case sensitive, so we need to make the code for roles handle this, so that "CO", "co", "Co" all enable the dropdown for the CO for the PCOL to be filled correctly and for it to see a user's role when they log in. Same with other roles, like Reviewer and Distributor.